spec.emu

```diff
@@ -835,12 +1711,179 @@ contributors: Luca Casonato, Guy Bedford, Nicolò Ribaudo
       <emu-alg>
         1. Return a new empty List.
       </emu-alg>
-      <emu-grammar><ins>ImportDeclaration : `import` `module` ImportedBinding FromClause `;`</ins></emu-grammar>
+      <emu-grammar><ins>ImportDeclaration : `import` `source` ImportedBinding FromClause `;`</ins></emu-grammar>
       <emu-alg>
         1. <ins>Let _localName_ be the sole element of BoundNames of |ImportedBinding|.</ins>
-        1. <ins>Let _entry_ be the ImportEntry Record { [[ModuleRequest]]: _module_, [[ImportName]]: ~module~, [[LocalName]]: _localName_ }.</ins>
-        1. <ins>Return &laquo; _entry_ &raquo;.</ins>
+        1. <ins>Let _entry_ be the ImportEntry Record { [[ModuleRequest]]: _module_, [[ImportName]]: ~source~, [[LocalName]]: _localName_ }.</ins>
+        1. <ins>Return « _entry_ »</ins>.
+      </emu-alg>
+    </emu-clause>
+  </emu-clause>
+</emu-clause>
+
+<emu-clause id="sec-global-object" number=19>
+  <h1>The Global Object</h1>
+
+  <emu-clause id="sec-constructor-properties-of-the-global-object">
+    <h1>Constructor Properties of the Global Object</h1>
+
+    <emu-clause id="sec-constructor-properties-of-the-global-object-module-source">
+      <h1>ModuleSource ( . . . )</h1>
+      <p>See <emu-xref href="#sec-modulesource-constructor"></emu-xref>.</p>
+    </emu-clause>
+  </emu-clause>
+</emu-clause>
+
+<emu-clause id="sec-reflection" number=28>
+  <h1>Reflection</h1>
+  <emu-clause id="sec-module-source-objects">
+    <h1><ins>Module Source Objects</ins></h1>
+    <p>Module Source Objects represent modules in their source import phase, which are not linked, instantiated or executed.</p>
+    <p>All Module Source Objects, including ModuleSource and host-defined types, have a prototype of %AbstractModuleSource%.prototype and provide a [[ModuleSourceRecord]] internal slot.</p>
+    <p>The ModuleSource Object represents the standard representation of the source of a Source Text Module Module record. Hosts may define their own Module Source Objects for other module record types.</p>
+
+    <emu-clause id="sec-abstractmodulesource-constructor">
+      <h1>The AbstractModuleSource Constructor</h1>
+      <p>The AbstractModuleSource constructor:</p>
+      <ul>
+        <li>is <dfn>%AbstractModuleSource%</dfn>.</li>
+        <li>is not a global object.</li>
+        <li>throws when called as a constructor.</li>
+        <li>also throws when called as a function rather than as a constructor.</li>
+        <li>has a *"length"* property whose value is *0*<sub>𝔽</sub>.</li>
+        <li>has a [[Prototype]] internal slot whose value is %Function.prototype%.</li>
```

Comment:
The definition of `[[Prototype]]` is duplicated here and in _"Properties of the %AbstractModuleSource% Intrinsic Object"_.

Given that this object cannot be really used as a constructor, I would just copy what `%TypedArray%` does (https://tc39.es/ecma262/#sec-%typedarray%-intrinsic-object).